apps/dashboard/src/components/pay/PayAnalytics/components/TotalVolumePieChart.tsx

```diff
@@ -0,0 +1,212 @@
+import { cn } from "@/lib/utils";
+import { Cell, Pie, PieChart } from "recharts";
+import { SkeletonContainer } from "../../../../@/components/ui/skeleton";
+import { usePayVolume } from "../hooks/usePayVolume";
+import { NoDataAvailable, chartHeight } from "./common";
+
+type VolData = {
+  name: string;
+  amount: number;
+  color: string;
+};
+
+type UIQueryData = {
+  totalAmount: number;
+  cryptoTotalUSD: number;
+  fiatTotalUSD: number;
+};
+
+export function TotalVolumePieChart(props: {
+  clientId: string;
+  from: Date;
+  to: Date;
+}) {
+  const volumeQuery = usePayVolume({
+    clientId: props.clientId,
+    from: props.from,
+    intervalType: "day",
+    to: props.to,
+  });
+
+  function getUIData(): {
+    data?: UIQueryData;
+    isError?: boolean;
+    isLoading?: boolean;
+  } {
+    if (volumeQuery.isLoading) {
+      return { isLoading: true };
+    }
+
+    if (volumeQuery.isError) {
+      return { isError: true };
+    }
+
+    if (volumeQuery.data.aggregate.sum.succeeded.amountUSDCents === 0) {
+      return { isError: true };
+    }
+
+    const cryptoTotalUSD = Math.ceil(
+      volumeQuery.data.aggregate.buyWithCrypto.succeeded.amountUSDCents / 100,
+    );
+    const fiatTotalUSD = Math.ceil(
+      volumeQuery.data.aggregate.buyWithFiat.succeeded.amountUSDCents / 100,
+    );
+
+    const totalAmount = cryptoTotalUSD + fiatTotalUSD;
+
+    const data: UIQueryData = {
+      totalAmount,
+      cryptoTotalUSD,
+      fiatTotalUSD,
+    };
+
+    return { data };
+  }
+
+  const uiData = getUIData();
```

Comment:
this pattern is so odd lol

Reply:
It's basically a mapping of query where I am modifying the data and treating certain data as error